src/app/shared/pipes/ref-data-decode/ref-data-decode.pipe.ts

```diff
@@ -21,20 +22,28 @@ export class RefDataDecodePipe implements PipeTransform, OnDestroy {
   transform(
     value: string | number | undefined,
     resourceType: string | undefined,
-    decodeKey: string = 'description'
+    decodeKey: string | number = 'description'
   ): Observable<string | number | undefined> {
     if (!resourceType || !value) {
       return of(value);
     }
 
-    this.store.dispatch(fetchReferenceDataByKey({ resourceType: resourceType as ReferenceDataResourceType, resourceKey: value }));
+    this.store.dispatch(fetchReferenceData({ resourceType: resourceType as ReferenceDataResourceType }));
+    this.store.dispatch(
+      fetchReferenceDataByKeySearch({ resourceType: (resourceType + '#AUDIT') as ReferenceDataResourceType, resourceKey: value + '#' })
+    );
 
-    return this.store.select(selectReferenceDataByResourceKey(resourceType as ReferenceDataResourceType, value)).pipe(
+    return merge(
+      this.store.select(selectReferenceDataByResourceKey(resourceType as ReferenceDataResourceType, value)),
+      this.store.select(selectSearchReturn((resourceType + '#AUDIT') as ReferenceDataResourceTypeAudit))
+    ).pipe(
       takeUntil(this.destroy$),
-      map(refDataItem => {
-        if (!refDataItem) return value;
-
-        return refDataItem[decodeKey as keyof ReferenceDataModelBase] ?? value;
+      map(item => {
+        if (Array.isArray(item)) {
+          return item[0][decodeKey as keyof ReferenceDataModelBase] ?? value;
+        } else {
+          return item?.[decodeKey as keyof ReferenceDataModelBase] ?? value;
```

Comment:
Yep, if it's an array that has been returned first then there's no current ref data and we need to show the audited/deleted data.
So `combineLatest` needed both observables to emit a value, we tried the zip operator too. If the audit history doesn't produce any values then the tests fail, as searchReturn is undefined.

Reply:
So, from what you're saying this is prioritising the displaying of audited data over non audited data? That doesn't seem right. Surely we want to see the current one as a priority.

Reply:
So there will only be an array emitted first if there's no current ref data, if there is current ref data then that won't be an array and it will move on to the next thing.